ethsigner/core/src/main/java/tech/pegasys/ethsigner/core/requesthandler/sendtransaction/transaction/BesuPrivateNonceProvider.java

```diff
@@ -12,54 +12,37 @@
  */
 package tech.pegasys.ethsigner.core.requesthandler.sendtransaction.transaction;
 
+import tech.pegasys.ethsigner.core.jsonrpc.JsonRpcRequest;
 import tech.pegasys.ethsigner.core.requesthandler.sendtransaction.NonceProvider;
 
-import java.io.IOException;
 import java.math.BigInteger;
 
-import org.apache.logging.log4j.LogManager;
-import org.apache.logging.log4j.Logger;
-import org.web3j.protocol.besu.Besu;
-import org.web3j.protocol.core.Request;
-import org.web3j.protocol.core.methods.response.EthGetTransactionCount;
 import org.web3j.utils.Base64String;
 
 public class BesuPrivateNonceProvider implements NonceProvider {
 
-  private static final Logger LOG = LogManager.getLogger();
-
-  private final Besu besu;
+  private final VertxNonceRequestTransmitter vertxNonceRequestTransmitter;
   private final String accountAddress;
   private final Base64String privacyGroupId;
 
-  BesuPrivateNonceProvider(
-      final Besu besu, final String accountAddress, final Base64String privacyGroupId) {
-    this.besu = besu;
+  public BesuPrivateNonceProvider(
+      final String accountAddress,
+      final Base64String privacyGroupId,
+      final VertxNonceRequestTransmitter vertxNonceRequestTransmitter) {
+    this.vertxNonceRequestTransmitter = vertxNonceRequestTransmitter;
     this.accountAddress = accountAddress;
     this.privacyGroupId = privacyGroupId;
   }
 
   @Override
   public BigInteger getNonce() {
-    return getNonceFromClient();
+    final JsonRpcRequest request = generateRequest();
+    return vertxNonceRequestTransmitter.requestNonce(request);
   }
 
-  private BigInteger getNonceFromClient() {
-
-    final Request<?, EthGetTransactionCount> request =
-        besu.privGetTransactionCount(accountAddress, privacyGroupId);
-    try {
-      LOG.debug(
-          "Retrieving Transaction count from eea provider for {} with privacy group id {}",
-          accountAddress,
-          privacyGroupId);
-      final EthGetTransactionCount count = request.send();
-      final BigInteger transactionCount = count.getTransactionCount();
-      LOG.trace("Reported transaction count for {} is {}", accountAddress, transactionCount);
-      return transactionCount;
-    } catch (final IOException e) {
-      LOG.info("Failed to determine nonce from downstream handler.", e);
-      throw new RuntimeException("Unable to determine nonce from eea provider.", e);
-    }
+  protected JsonRpcRequest generateRequest() {
```

Comment:
can this be initialized once in constructor? Assuming that accountAddress and privacygroupid would remain final for the duration of this class.